**ABSTRACT:**
This research investigates child malnutrition trends across countries using the UNICEF Joint Child Malnutrition Estimates dataset. Key indicators such as stunting, wasting, overweight, and underweight among children under five years of age are analyzed over time and across regions. The study aims to identify patterns, regional disparities, and progress toward Sustainable Development Goals related to child health and nutrition. Data cleaning, visualization, and basic statistical analysis are employed to extract insights, providing a clear understanding of global malnutrition challenges and highlighting areas requiring policy attention.

In [1]:
import os
import pandas as pd

# Define the file path
file_path = os.path.join("..", "data", 'raw', 'main', "JME_Country_Estimates_April_2021.xlsx")

# Load the sheet names
xls = pd.ExcelFile(file_path)
print("Sheets available:\n", xls.sheet_names, "\n")

# Preview the first 5 rows of each sheet
for sheet in xls.sheet_names:
    print(f"--- Preview of sheet: {sheet} ---")
    df = pd.read_excel(file_path, sheet_name=sheet)
    print(df.head(5))
    print("\n")


Sheets available:
 ['Cover Page', 'Survey Estimates', 'Notes', 'Excluded Sources', 'Stunting Proportion (Model)', 'Stunting Numb Affected(Model)', 'Overweight Proportion (Model)', 'Overweight Numb Affected(Model)'] 

--- Preview of sheet: Cover Page ---
         Unnamed: 0 Unnamed: 1 Unnamed: 2                     Unnamed: 3  \
0               NaN        NaN        NaN                            NaN   
1               NaN        NaN        NaN                            NaN   
2               NaN        NaN        NaN                            NaN   
3  data.unicef.org         NaN        NaN  who.int/nutgrowthdb/estimates   
4               NaN        NaN        NaN                            NaN   

  Unnamed: 4  Unnamed: 5  Unnamed: 6  Unnamed: 7          Unnamed: 8  
0        NaN         NaN         NaN         NaN                 NaN  
1        NaN         NaN         NaN         NaN                 NaN  
2        NaN         NaN         NaN         NaN                 NaN  
3    

From this preview, the best approach for this research would be using the 'Survey Estimates' sheet, as it seems to contain the country-wise malnutrition data...

In [2]:
# Load the 'Survey Estimates' sheet
df = pd.read_excel(file_path, sheet_name='Survey Estimates')

# Quick look
print(df.shape)
df.head()

(997, 29)


,ISO code,Country and areas,Survey year,Year*,United Nations Region,United Nations Sub-Region,SDG Region,UNICEF Region,UNICEF Sub-Region,WHO Region,...,Wasting,Overweight,Stunting,Underweight,Notes,Report Author,Source,Short Source,Fieldwork Period,U5 Population ('000s)
0,AFG,AFGHANISTAN,1997,1997,Asia,Southern Asia,Central Asia and Southern Asia,SA,SA,EMRO,...,18.2,6.5,53.2,44.9,cw 10,CIET International,Afghanistan 1997 multiple indicator baseline (...,MICS,"January,1997-May,1997",3838.876953
1,AFG,AFGHANISTAN,2004,2004,Asia,Southern Asia,Central Asia and Southern Asia,SA,SA,EMRO,...,8.6,4.6,59.3,32.9,w 11,"Ministry of Public Health (Afghanistan), UNICE...",Summary report of the national nutrition surve...,NNS,"May,2004-June,2004",4789.353027
2,AFG,AFGHANISTAN,2013,2013,Asia,Southern Asia,Central Asia and Southern Asia,SA,SA,EMRO,...,9.5,5.3,40.4,24.6,rs,"Ministry of Public Health, UNICEF and the Aga ...",Afghanistan National Nutrition Survey 2013,NNS,"May,2013-October,2013",5444.572754
3,AFG,AFGHANISTAN,2018,2018,Asia,Southern Asia,Central Asia and Southern Asia,SA,SA,EMRO,...,5.1,4.1,38.2,19.1,rs,KIT Royal Tropical Institute,Afghanistan Health Survey 2018,Other,"March,2018-November,2018",5601.442871
4,AGO,ANGOLA,1996,1996,Africa,Middle Africa,Sub-Saharan Africa,SSA,ESA,AFRO,...,7.7,1.7,61.1,36.2,rs,Instituto Nacional de Estatistica - Gabinete d...,Inquerito de indicadores multiplos (MICS) 1996...,MICS,"August,1996-December,1996",2749.750000


In [3]:
# Drop fully empty columns
df = df.dropna(axis=1, how='all')

# Drop fully empty rows
df = df.dropna(axis=0, how='all')

# Check the cleaned head
df.head()
print(df.shape)

(997, 29)


In [4]:
# Make a copy first
df_clean = df.copy()

# Clean column names
df_clean.columns = df_clean.columns.str.strip()  # Remove spaces at start/end
df_clean.columns = df_clean.columns.str.replace('*', '', regex=False)  # Remove asterisks
df_clean.columns = df_clean.columns.str.replace(' ', '_')  # Replace spaces with underscores
df_clean.columns = df_clean.columns.str.replace('-', '_')  # Replace dashes with underscores
df_clean.columns = df_clean.columns.str.replace("'", '', regex=False)  # Remove single quotes
df_clean.columns = df_clean.columns.str.replace("(", '', regex=False)  # Remove single quotes
df_clean.columns = df_clean.columns.str.replace(")", '', regex=False)  # Remove single quotes

# Check cleaned column names
df_clean.columns


Index(['ISO_code', 'Country_and_areas', 'Survey_year', 'Year',
       'United_Nations_Region', 'United_Nations_Sub_Region', 'SDG_Region',
       'UNICEF_Region', 'UNICEF_Sub_Region', 'WHO_Region',
       'World_Bank_Income_Classification', 'World_Bank_Region', 'LDC', 'LIFD',
       'LLDC_or_SIDS', 'UNICEF_Survey_ID', 'WHO_Global_Database_Number',
       'WHZ_Survey_Sample_N', 'Severe_Wasting', 'Wasting', 'Overweight',
       'Stunting', 'Underweight', 'Notes', 'Report_Author', 'Source',
       'Short_Source', 'Fieldwork_Period', 'U5_Population_000s'],
      dtype='object')

In [5]:
#Check for missing values
df_clean.isnull().sum()

ISO_code                              0
Country_and_areas                     0
Survey_year                           0
Year                                  0
United_Nations_Region                 0
United_Nations_Sub_Region             0
SDG_Region                            0
UNICEF_Region                        12
UNICEF_Sub_Region                    12
WHO_Region                            7
World_Bank_Income_Classification      0
World_Bank_Region                     0
LDC                                 639
LIFD                                568
LLDC_or_SIDS                        639
UNICEF_Survey_ID                      0
WHO_Global_Database_Number            0
WHZ_Survey_Sample_N                  68
Severe_Wasting                      213
Wasting                              45
Overweight                          125
Stunting                             35
Underweight                          22
Notes                               477
Report_Author                         1


**Insights till now:**

Minor missing values (can fill or drop a few rows):

WHO_Region (7)

Stunting (35), Underweight (22), Wasting (45), Overweight (125)

U5_Population_000s (3)

___________________________________________________________________

Moderate missing values (need consideration):

Fieldwork_Period (132)

UNICEF_Region / UNICEF_Sub_Region (12)

___________________________________________________________________

High missing values (may not be useful for analysis):

LDC (639), LIFD (568), LLDC_or_SIDS (639)

Notes (477)

___________________________________________________________________

In [6]:
#Selecting Relevant Columns
cols_to_keep = [
    'ISO_code', 'Country_and_areas', 'Year', 'SDG_Region', 'UNICEF_Region', 
    'WHO_Region', 'Wasting', 'Stunting', 'Overweight', 'Underweight', 'U5_Population_000s'
]

df_analysis = df_clean[cols_to_keep].copy()

In [7]:
#Check missing values in the new subset dataframe

print(df_analysis.isnull().sum())

ISO_code                0
Country_and_areas       0
Year                    0
SDG_Region              0
UNICEF_Region          12
WHO_Region              7
Wasting                45
Stunting               35
Overweight            125
Underweight            22
U5_Population_000s      3
dtype: int64


In [8]:
#Handling Missing Values

#For numeric columns, we can fill missing values with the mean or median
numeric_cols = ['Wasting', 'Stunting', 'Overweight', 'Underweight', 'U5_Population_000s']
for col in numeric_cols:
    df_analysis[col] = df_analysis[col].fillna(df_analysis[col].median())


#For categorical columns, we can fill missing values with 'Unknown' or the mode
categorical_cols = ['UNICEF_Region', 'WHO_Region']
for col in categorical_cols:
    df_analysis[col] = df_analysis[col].fillna('Unknown')

#Verify no missing values remain
print(df_analysis.isnull().sum())

ISO_code              0
Country_and_areas     0
Year                  0
SDG_Region            0
UNICEF_Region         0
WHO_Region            0
Wasting               0
Stunting              0
Overweight            0
Underweight           0
U5_Population_000s    0
dtype: int64


In [9]:
# Define path to save cleaned dataset
processed_path = os.path.join("..", "data", "processed_and_cleaned", "child_malnutrition_cleaned.csv")

# Save the cleaned dataframe
df_analysis.to_csv(processed_path, index=False)

print(f"Cleaned dataset saved to: {processed_path}")


Cleaned dataset saved to: ..\data\processed_and_cleaned\child_malnutrition_cleaned.csv
